packages/venia-ui/lib/drivers/adapter.js

```diff
@@ -86,15 +90,103 @@ const VeniaAdapter = props => {
         return null;
     }
 
+    let storeCodeRouteHandler = '';
+    const browserRouterProps = {};
+    if (process.env.USE_STORE_CODE_IN_URL === 'true') {
+        const storeCode = storage.getItem('store_view_code') || STORE_VIEW_CODE;
+        browserRouterProps.basename = `/${storeCode}`;
+        const storeCodes = [];
+        const storeCurrencies = {};
+        if (Array.isArray(AVAILABLE_STORE_VIEWS)) {
+            AVAILABLE_STORE_VIEWS.forEach(store => {
+                storeCodes.push(store.code);
+                storeCurrencies[store.code] =
+                    store.default_display_currency_code;
+            });
+        }
+
+        /**
+         * The current store code won't be matched as it's included as the basename, if we match another store
+         * we need to change the current store.
+         */
+        storeCodeRouteHandler = (
+            <Switch>
+                <Route path={`/:storeCode(${storeCodes.join('|')})?`}>
+                    {({ match }) => {
+                        if (
+                            match.params.storeCode &&
```

Comment:
When I clear my cache and hit https://magento-venia-concept-mhkx8.local.pwadev:8389/fr/ I am seeing `match.params.storeCode === 'fr'` but when I refresh after it loads it's undefined. Why?

Reply:
Because the base name is including `fr`, meaning fr is no longer considered in the `<Route />` match.

Reply:
I see - it is because the basename has changed, so once `fr` is the basename it doesn't match because it is part of the base.